cli/ballerina-launcher/src/main/java/org/ballerinalang/launcher/Main.java

```diff
@@ -186,42 +177,48 @@ private static String prepareCompilerErrorMessage(String message) {
         return "ballerina: " + LauncherUtils.makeFirstLetterLowerCase(message);
     }
 
+    private static String getFirstUnknownArg(String errorMessage) {
+        String optionsString = errorMessage.split(":")[1];
```

Comment:
Will this never produce array index out of bound?

Reply:
Ideally it shouldn't, since this method is called only when an UnmatchedArgumentException is thrown and here we would expect a ":" followed by one or more arguments. https://github.com/remkop/picocli/blob/master/src/main/java/picocli/CommandLine.java#L8995.